### PR TITLE
feature(frontend) add email verification to protected renew app

### PR DESCRIPTION
**Pull request by Fbasham**
### Description
adds email verification to the protected renew app (not the ITA portion, this will be addressed in a separate PR).  I'm sure there are issues, or edge cases that I haven't found yet, please test.

### Related Azure Boards Work Items
AB#5951

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added/updated tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: frontend/app/routes/protected/renew/$id/confirm-email.tsx [hunk 1]
```diff
@@ -104,6 +104,10 @@ export async function action({ context: { appContainer, session }, params, reque
   const idToken: IdToken = session.get('idToken');
   appContainer.get(TYPES.domain.services.AuditService).createAudit('update-data.renew.confirm-email', { userId: idToken.sub });
 
+  if (parsedDataResult.data.email && parsedDataResult.data.email !== state.contactInformation?.email) {
+    return redirect(getPathById('protected/renew/$id/verify-email', params));
+  }
```
Comment (nicholas-ly) [inline frontend/app/routes/protected/renew/$id/confirm-email.tsx:109]: At this point, we should be sending the initial verification code email and saving it to the state object.

We should also check if the `emailVerified` field is set in state and redirect to `/verify-email` if it has not. It is currently possible to enter an email, then hit back on `/verify-email`, and hit save (with no changes to the email) on `/confirm-email` and be redirect to the review page without ever verifying the email.

File: frontend/app/routes/protected/renew/$id/verify-email.tsx [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+          </LoadingButton>
+        </fieldset>
+        <div className="flex flex-row-reverse flex-wrap items-center justify-end gap-3">
+          <LoadingButton variant="primary" id="continue-button" name="_action" value={FORM_ACTION.submit} loading={isSubmitting} endIcon={faChevronRight} data-gc-analytics-customclick="ESDC-EDSC:CDCP Online Application Form-Adult:Continue - Verify email">
+            {t('protected-renew:verify-email.continue')}
+          </LoadingButton>
+          <ButtonLink id="back-button" routeId="protected/renew/$id/confirm-email" params={params} disabled={isSubmitting} startIcon={faChevronLeft} data-gc-analytics-customclick="ESDC-EDSC:CDCP Online Application Form-Adult:Back - Verify email click">
+            {t('protected-renew:verify-email.back')}
+          </ButtonLink>
+        </div>
```
Comment (nicholas-ly) [inline frontend/app/routes/protected/renew/$id/verify-email.tsx:241]: Something to look at in a future PR is adding edit mode. One design that was suggested was having the `/confirm-communication-preference` route appear for all renewing primary applicants (similar to `/dental-insurance`). If the applicant chooses to communicate via email, then they will be redirected to the email verification page.

(I'll create an ADO work item in the future when we give this more thought/if this makes sense)

File: frontend/app/routes/protected/renew/$id/confirm-email.tsx (hunk 1 above)
Comment (Fbasham) [reply frontend/app/routes/protected/renew/$id/confirm-email.tsx]: I just pushed a change that I believe fixes this issue.  The ITA portion of the protected renew app still needs to be done and I haven't given much thought to it, or if additional state variables and logic will need to be added or changed similar to the apply app(s).

File: frontend/app/routes/protected/renew/$id/confirm-email.tsx [hunk 3]
```diff
@@ -98,12 +100,33 @@ export async function action({ context: { appContainer, session }, params, reque
...
+    saveProtectedRenewState({
+      params,
+      request,
+      session,
+      state: {
+        verifyEmail: {
+          verificationCode,
+          verificationAttempts: 0,
+        },
+        emailVerified: false, // Reset emailVerified when requesting a new code
```
Comment (nicholas-ly) [inline frontend/app/routes/protected/renew/$id/confirm-email.tsx:123]: There is an edge case where if the applicant previously verified their email, then goes to change their email and does not verify it (by clicking "Back" and "Cancel"), `emailVerified` would be `false`. It should be `true` in this case since their previous email has been verified. 

This is handled in the apply flows with an `isNewEmail` local variable.

File: frontend/app/routes/protected/renew/$id/confirm-email.tsx [hunk 4]
```diff
@@ -98,12 +100,33 @@ export async function action({ context: { appContainer, session }, params, reque
...
+    state: { emailPreValidation: parsedDataResult.data },
+    // state: { contactInformation: { ...state.contactInformation, ...parsedDataResult.data } },
   });
 
   const idToken: IdToken = session.get('idToken');
   appContainer.get(TYPES.domain.services.AuditService).createAudit('update-data.renew.confirm-email', { userId: idToken.sub });
 
+  if (parsedDataResult.data.email && parsedDataResult.data.email !== state.contactInformation?.email) {
+    // Create a new verification code and store the code in session
+    const verificationCode = verificationCodeService.createVerificationCode('anonymous');
```
Comment (nicholas-ly) [inline frontend/app/routes/protected/renew/$id/confirm-email.tsx:112]: Just noticed this should be changed for all `protected` pages (can be done in another PR - will add an ADO work item):
```suggestion
    const verificationCode = verificationCodeService.createVerificationCode(idToken.sub);
```

File: frontend/app/routes/protected/renew/$id/confirm-email.tsx (hunk 3 above)
Comment (Fbasham) [reply frontend/app/routes/protected/renew/$id/confirm-email.tsx:126]: i took a bit of a different approach, hopefully it works as expected 😬

File: frontend/app/routes/protected/renew/$id/confirm-email.tsx (hunk 4 above)
Comment (Fbasham) [reply frontend/app/routes/protected/renew/$id/confirm-email.tsx:115]: since I picked up the ADO ticket, I'll address in a separate PR

**Review by nicholas-ly — APPROVED** [review]
